kodi_addon_checker/check_string.py

```diff
@@ -38,3 +39,54 @@ def find_blacklisted_strings(report: Report, addon_path: str, problems: list, wa
     for result in handle_files.find_in_file(addon_path, warnings, file_types):
         report.add(Record(WARNING, "Found blacklisted term %s in file %s:%s (%s)"
                           % (result["term"], result["searchfile"], result["linenumber"], result["line"])))
+
+
+def check_for_invalid_strings_po(report: Report, file_index: list):
+    """Validate strings.po files
+        :file_index: list having names and path of all the files present in addon
+    """
+    po_file_index = [f for f in file_index if f["name"] == "strings.po"]
+    report_made = False
+
+    for po_file in po_file_index:
+        full_path = os.path.join(po_file["path"], po_file["name"])
+
+        with open(full_path, "r", encoding="utf-8") as f:
+            contents = f.read()
```

Comment:
If the file is not saved with utf-8 encoding, open raises an exception.
```suggestion
            try:
                contents = f.read()
            except UnicodeDecodeError:
                report_made = True
                report.add(Record(PROBLEM, "Invalid PO file %s: File is not saved with UTF-8 encoding" % (relative_path(full_path))))
                continue
```

Reply:
Added